Code/Framework/AzCore/AzCore/std/containers/span.inl

```diff
@@ -24,7 +24,7 @@ namespace AZStd
     }
 
     template <class Element>
-    inline constexpr span<Element>::span(pointer first, const_pointer last)
+    inline constexpr span<Element>::span(pointer first, pointer last)
```

Comment:
Was this a `const_pointer` previously for a reason or was it unintentional? (I see the `first` param was already a `pointer`)

Reply:
That was unintentional. I must have missed the second parameter when I was making the change.

Reply:
Yeah, it looked like a bug, and it was causing my code not to compile since I was apparently hitting that constructor, so I fixed it. :)